nsxt/resource_nsxt_policy_context_profile_test.go

```diff
@@ -0,0 +1,317 @@
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.value.2076247700", "SSL"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.sub_attribute.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.sub_attribute.250747496.cifs_smb_version.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.sub_attribute.250747496.tls_cipher_suite.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.sub_attribute.250747496.tls_version.#", "3"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.sub_attribute.250747496.tls_version.2416475543", "TLS_V12"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.sub_attribute.250747496.tls_version.2721980181", "TLS_V10"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.2652970086.sub_attribute.250747496.tls_version.645341863", "SSL_V3"),
+				),
+			},
+			{
+				Config: testAccNsxtPolicyContextProfileTemplate(updatedName, updatedAttributes),
+				Check: resource.ComposeTestCheckFunc(
+					testAccNsxtPolicyContextProfileExists(testResourceName),
+					resource.TestCheckResourceAttr(testResourceName, "display_name", updatedName),
+					resource.TestCheckResourceAttr(testResourceName, "description", "Acceptance Test"),
+					resource.TestCheckResourceAttrSet(testResourceName, "nsx_id"),
+					resource.TestCheckResourceAttrSet(testResourceName, "path"),
+					resource.TestCheckResourceAttrSet(testResourceName, "revision"),
+					resource.TestCheckResourceAttr(testResourceName, "tag.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "domain_name.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "url_category.#", "0"),
+					resource.TestCheckResourceAttrSet(testResourceName, "app_id.832058275.is_alg_type"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.value.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.value.1810355442", "CIFS"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.sub_attribute.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.sub_attribute.2393957168.cifs_smb_version.#", "2"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.sub_attribute.2393957168.tls_cipher_suite.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.sub_attribute.2393957168.tls_version.#", "0"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.sub_attribute.2393957168.cifs_smb_version.3398512106", "CIFS_SMB_V1"),
+					resource.TestCheckResourceAttr(testResourceName, "app_id.832058275.sub_attribute.2393957168.cifs_smb_version.3787226665", "CIFS_SMB_V2"),
+				),
+			},
```

Comment:
can you please add another step here where sub attribute is removed, in order to make sure empty list is not ignores by patch?

Reply:
Sure I will add these two actual steps.